core/src/main/java/org/elasticsearch/index/fielddata/UidIndexFieldData.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.fielddata;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.fielddata.IndexFieldData.XFieldComparatorSource.Nested;
+import org.elasticsearch.index.fielddata.fieldcomparator.BytesRefFieldComparatorSource;
+import org.elasticsearch.index.fielddata.plain.AbstractAtomicOrdinalsFieldData;
+import org.elasticsearch.index.mapper.UidFieldMapper;
+import org.elasticsearch.search.MultiValueMode;
+
+import java.io.IOException;
+
+/** Fielddata view of the _uid field on indices that do not index _uid but _id.
+ *  It gets fielddata on the {@code _id field}, which is in-memory since the _id
+ *  field does not have doc values, and prepends {@code ${type}#} to all values.
+ *  Note that it does not add memory compared to what fielddata on the _id is
+ *  already using: this is just a view.
+ *  TODO: Remove fielddata access on _uid and _id, or add doc values to _id.
```

Comment:
this TODO worries me a bit, I think we use this access by default for instance when we slice scrolls? Should we add docvalues to `_id` as well? I wonder if we should open a new issue and make it a blocker to ensure we make a call here?